### PR TITLE
refactor: treeshakable kind enum

### PR DESCRIPTION
**Pull request by jasonkuhrt**
closes https://github.com/graphql/graphql-js/issues/4253

### CODE REVIEW

**Comment by jasonkuhrt** [comment]
Duplicate of https://github.com/graphql/graphql-js/pull/4268.

**Comment by jasonkuhrt** [comment]
@JoviDeCroock needed to focus on other things so #4268 was closed. I am re-opening this PR therefore to finish that work.

File: integrationTests/ts/package.json [hunk 1]
```diff
@@ -8,11 +8,6 @@
   "dependencies": {
     "graphql": "file:../graphql.tgz",
     "graphql-esm": "file:../graphql-esm.tgz",
-    "typescript-4.4": "npm:typescript@4.4.x",
-    "typescript-4.5": "npm:typescript@4.5.x",
-    "typescript-4.6": "npm:typescript@4.6.x",
-    "typescript-4.7": "npm:typescript@4.7.x",
-    "typescript-4.8": "npm:typescript@4.8.x",
```
Comment (jasonkuhrt) [inline integrationTests/ts/package.json:15]: Regarding this, was already discussed at https://github.com/graphql/graphql-js/pull/4268#issuecomment-2445185032.

File: src/language/__tests__/predicates-test.ts
Comment (jasonkuhrt) [inline src/language/__tests__/predicates-test.ts:1]: The `Kind` enum member order is now alphabetical per how ESM namespace keys are serialized (I guess).

I've updated the orders in this tests suite. I assume their order did not matter before.

File: src/language/__tests__/kind-test.ts
Comment (jasonkuhrt) [inline src/language/__tests__/kind-test.ts:1]: I've added some sanity checks here. Let me know if there is an established type testing approach in the library. For now I did something quick and simple.

File: src/language/__tests__/predicates-test.ts
Comment (yaacovCR) [reply src/language/__tests__/predicates-test.ts:1]: If it did, we are targeting v17 so this would be ok

**Review by yaacovCR — APPROVED** [review]
Do we want to take this approach for all enums?

**Comment by jasonkuhrt** [comment]
Yeah I think there's nothing to lose by doing that and it would be a better DX (e.g. all enum have type level name spaces, consistent).

If you agree then I can update this PR to change them all.

**Review by JoviDeCroock — APPROVED** [review]
I think the odds of other enums tree-shaking is a lot lower and the amount of code we add in this approach is quite high. The others like DirectiveLocation, ... are more geared for server-side code either way. I don't merging as is as the bundle impact is reduced by moving off of TS enums

**Comment by jasonkuhrt** [comment]
Right now I don't have a strong preference. I consider the type level symmetry aspect bike shedding territory. Happy to throw my labour at this so no issue there.

**Pull request merged**